src/test/java/com/myhome/services/unit/CommunitySDJpaServiceTest.java

```diff
@@ -337,13 +338,17 @@ void deleteCommunity() {
 
     given(communityRepository.findByCommunityId(TEST_COMMUNITY_ID))
         .willReturn(Optional.of(testCommunity));
+    testCommunityHouses.forEach(house -> {
+      given(communityHouseRepository.findByHouseId(house.getHouseId()))
+        .willReturn(house);
+    });
 
     // when
     boolean communityDeleted = communitySDJpaService.deleteCommunity(TEST_COMMUNITY_ID);
 
     // then
     assertTrue(communityDeleted);
-    verify(communityRepository).findByCommunityId(TEST_COMMUNITY_ID);
+    verify(communityRepository, times(3)).findByCommunityId(TEST_COMMUNITY_ID);
```

Comment:
I guess community should be passed to `removeHouseFromCommunityByHouseId` method instead of just id since we dont want to make `findByCommunityId` twice for same Id

Reply:
Ya I was thinking that. This would require a change to code that calls that method too to pass the Community in. Or else we could have a private method and the public facing method with community id then just calls that private method. Our delete community method could then call the private one?

Reply:
Let's just switch the method arguments from:
`removeHouseFromCommunityByHouseId(String communityId, String houseId)`

to:

`removeHouseFromCommunityByHouseId(Community community, String houseId)`

that should not be a big effort to change that but will do significant code adjustment

Reply:
It will need to refactor the community unit test and possible other code if we use that approach rather than a public facing function with the ID. But I can do that